File: geradorwebapp/Gerar-APP/src/app/_models/TipoVariavel.ts
import { PLC } from './PLC';
import { Ambev } from './Ambev';
import { Historian } from './Historian';
import { LmsTipoVariavel } from './LmsTipoVariavel';

export interface TipoVariavel {
      id: number;
      descricao: string;
      plcs: PLC[];
      ambev: Ambev[];
      historians: Historian[];
      lmsTipoVariavels: LmsTipoVariavel[];
}

File: gerdadorweb/Gerador.Domain/Entities/TipoVariavel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace Gerador.Domain.Entities
{
    public class TipoVariavel
    {
        [Required(ErrorMessage = "ID é obrigatório")]
        public int Id { get; set; }
        public string Descricao { get; set; }
        public virtual ICollection<Plc> Plcs { get; set; }
        public virtual ICollection<Ambev> Ambevs { get; set; }
        public virtual ICollection<Historian> Historians { get; set; }
        public virtual ICollection<Lms> LMSs { get; set; }
        public virtual ICollection<Lms> LMSsFT { get; set; }

        public virtual ICollection<IGS> IGSs { get; set; }
        public virtual ICollection<IGS> IGSsFT { get; set; }
    }

}
